lab-doug/app/component/note-item/index.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+require('./note-item.scss');
+
+const angular = require('angular');
+const widgetApp = angular.module('widgetApp');
+widgetApp.directive('appNoteItem', function(){
+  return {
+    restrict: 'E',
+    replace: true,
+    template: require('./note-item.html'),
+    controller: ['$log', 'noteService', NoteItemController],
```

Comment:
I would avoid defining controllers on the directive. Define the controller on the module, then specify it here as a string. If you do it that way, it's testable and can be injected w/ angular mocks. If you do it this way, you can't do that.